### PR TITLE
ConciseDateFormatter's offset string is correct on an inverted axis

### PR DESCRIPTION
**Pull request by OdileVidrine**
## PR summary
Fixes a bug with ConciseDateFormatter's offset string where the proper label is not shown if the axis is inverted. Closes https://github.com/matplotlib/matplotlib/issues/28481.

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by jklymak — APPROVED** [review]
this seems good.  Probably needs a short change note in api/next_api_changes/behavior, since this is actually a breaking change.  However, I don't think this should be made back-compatible, nor really deserving of an automated way to get the old behaviour.

File: lib/matplotlib/dates.py [hunk 1]
```diff
@@ -796,7 +796,12 @@ def format_ticks(self, values):
 
         if show_offset:
             # set the offset string:
-            self.offset_string = tickdatetime[-1].strftime(offsetfmts[level])
+            if (self._locator.axis and
+                    self._locator.axis.__name__ == ('xaxis' or 'yaxis')
```
[the remark this answers is not in the document]
Comment (OdileVidrine) [reply lib/matplotlib/dates.py:800]: thank you, I don't know how I missed that!

**Comment by rcomer** [comment]
Thanks @OdileVidrine, do you have time to write a change note as described at https://github.com/matplotlib/matplotlib/pull/28501#pullrequestreview-2156817282?

**Comment by OdileVidrine** [comment]
I wrote the change note.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @OdileVidrine!

**Pull request merged**